src/os/shared.rs

```diff
@@ -123,8 +124,11 @@ pub fn get_input(
     let write_to_stdout = create_write_to_stdout();
     let (on_winch, cleanup) = sigwinch();
     let dns_client = if resolve {
-        let (resolver, background) = dns::Resolver::new()?;
-        let dns_client = dns::Client::new(resolver, background)?;
+        let mut runtime = Runtime::new()?;
+        let resolver = runtime
+            .block_on(dns::Resolver::new(runtime.handle().clone()))
+            .unwrap();
```

Comment:
We can remove this `unwrap()` and use a question mark here

Reply:
I made this change but for my own edification, can you expand on the reasoning for it? I am certainly new to the idioms and best practices for Rust.

Reply:
Sure, I will try my best! 

`unwrap()` will try to access the value inside the `Result` and will panic in case of error. The question mark will desugar to a `match` statement that, in case of error, will return early from the function. Another nice thing from the `?` is that it will try to convert the unwrapped error into the error type in the return value of the signature, in case the latter implements [`From`](https://doc.rust-lang.org/std/convert/trait.From.html) for that error.

You can read more [here](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator) and [here](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/the-question-mark-operator-for-easier-error-handling.html).